### PR TITLE
Docker Config

### PR DESCRIPTION
**Pull request by blangley28**
Configurate docker to use environment variables to allow multiple instances of the same software. Update docker compose to start two instances. Renamed "medmorph" realm on Keycloak to"ehr" and added the "knowledgeartifact" realm. 

EHR and KA are running on pathways.mitre.org at:
EHR: http://pathways.mitre.org:8180/fhir
KA: http://pathways.mitre.org:8190/fhir

### CODE REVIEW

**Comment by blangley28** [comment]
Test

File: README.md [hunk 1]
```diff
@@ -1,39 +1,69 @@
...
 
 # Running Locally
 
-The easiest way to run this server is to use docker. First, clone this
-repository. Then, from the repository root run:
+The easiest way to run this server is to use docker.
 
 ```
-docker build -t medmorph_ehr .
+./build-docker-image.sh
```
Comment (dehall) [inline README.md]: super minor note - the .sh file is gone, this should be .bat now

File: README.md [hunk 2]
```diff
@@ -1,39 +1,69 @@
...
+Since this is a single repository and single docker image for multiple actors, each container should have its own properties. Right now the properties to configurate are `AUTH_SERVER_ADDRESS`, `SERVER_ADDRESS`, and `SERVER_TITLE`. These are set through enviornment variables of the same names. Using compose will automatically create all necessary servers.
 
 # Authorization
 
 This server is protected by the SMART Backend Authentication protocol. The admin token is `admin`. The OAuth URLS can be found at the `/metadata` or `/.well-known/smart-configuration` endpoints.
 
-Follow the [Register New Client](https://github.com/mcode/medmorph-ehr/wiki/Register-New-Client) wiki page to register a new client.
+Follow the [Register New Client](https://github.com/mcode/medmorph-fhir-server/wiki/Register-New-Client) wiki page to register a new client.
+
+# Hosted Server
```
Comment (dehall) [inline README.md]: Can we move this section to a wiki page? I feel like we don't want to list out the details of an internal instance on the public GitHub page. Not that it's a security risk, pathways.mitre.org isn't accessible externally at all, it's just not broadly meaningful. And then if we change things around like ports or URLs we don't have to issue PRs for the readme.

Also, probably worth pointing out that the ports and URLs are already modified in the docker-compose.yml on the server, so a `git pull` might fail if that file is changed in the repo. One option is to git stash, pull, then git stash pop, rebuild, etc.

Comment (blangley28) [reply README.md]: https://github.com/mcode/medmorph-fhir-server/wiki/MITRE-Internal-Server

**Review by dehall — APPROVED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -2,8 +2,22 @@ version: "3"
...
+  knowledge_artifact:
+    image: medmorph_fhir
+    container_name: knowledge_artifact
+    restart: on-failure
+    ports:
+      - "8090:8080"
+    environment:
+      - SERVER_ADDRESS=http://localhost:8090/fhir/
+      - AUTH_SERVER_ADDRESS=http://moonshot-dev.mitre.org:8090/auth/realms/knowledgeartifact/protocol/openid-connect/
+      - SERVER_TITLE='Knowledge Artifact Repository'
```
Comment (dehall) [inline docker-compose.yml]: Super super minor thing I noticed in the demo, but the name here seems to be double quoted in actual usage. (Which is strange, I thought you didn't need quotes in YML but putting them in anyway wouldn't break anything)
From http://pathways.mitre.org:8190/fhir/metadata:

```
{
  "resourceType": "CapabilityStatement",
  "title": "'Knowledge Artifact Repository'",
  "status": "active",
  "experimental": true,
  "date": "2021-01-21T14:45:28+00:00",
  ```

Comment (blangley28) [reply docker-compose.yml]: So it turns out unlike environment variables in bash you can have spaces and don't need to wrap it in quotes at all

**Review by KeeyanGhoreshi — APPROVED** [review]
looks fine to me

**Pull request merged**